### PR TITLE
implement interlace filter

### PR DESCRIPTION
**Pull request by wodka**
Idea from https://github.com/liip/LiipImagineBundle/issues/474

### CODE REVIEW

**Comment by lsmith77** [comment]
ping

**Comment by wodka** [comment]
is there an example for the test?

**Comment by makasim** [comment]
> is there an example for the test?

there are many test cases in Tests folder. You can look at them

**Comment by wodka** [comment]
ok :/ can you help me with this? how to I pass a Mock Object to check if it will be called to the InterlaceFilter?

```
Argument 1 passed to Liip\ImagineBundle\Imagine\Filter\Loader\InterlaceFilterLoader::load() must implement interface Imagine\Image\ImageInterface, instance of Mock_ImagineInterface_23d0ec1d given, called in /home/travis/build/liip/LiipImagineBundle/Tests/Imagine/Filter/Loader/InterlaceFilterLoaderTest.php on line 24 and defined
```

File: Tests/Imagine/Filter/Loader/InterlaceFilterLoaderTest.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use Liip\ImagineBundle\Tests\AbstractTest;
+
+/**
+ * @covers Liip\ImagineBundle\Imagine\Filter\Loader\InterlaceFilterLoader
+ */
+class InterlaceFilterLoaderTest extends AbstractTest
+{
+    public function testLoad()
+    {
+        $interlace = new InterlaceFilterLoader();
```
Comment (makasim) [inline Tests/Imagine/Filter/Loader/InterlaceFilterLoaderTest.php]: $loader

File: Tests/Imagine/Filter/Loader/InterlaceFilterLoaderTest.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+        $imagine = $this->createImagineMock();
```
Comment (makasim) [inline Tests/Imagine/Filter/Loader/InterlaceFilterLoaderTest.php]: not imagine but image.

**Comment by wodka** [comment]
thx - should I then squash the changes to one commit?

**Comment by makasim** [comment]
@wodka yes please (:

**Comment by makasim** [comment]
@wodka could you please add a func test like this one https://github.com/liip/LiipImagineBundle/blob/master/Tests/Functional/Imagine/FilterManagerTest.php#L8 to make sure the service is configured correctly and could be get from container.

**Comment by wodka** [comment]
should I also move the DataManagerTest & FilterManagerTest into the corresponding Subfolders?

**Comment by makasim** [comment]
> should I also move the DataManagerTest & FilterManagerTest into the corresponding Subfolders?

what do you mean? aren't they where they should be?

**Comment by wodka** [comment]
Right now they are in Imagine Folder, but should be in the corresponding subfolder.

Tests/Functional/Imagine/DataManagerTest -> Tests/Functional/Imagine/Data/DataManagerTest
Tests/Functional/Imagine/FilterManagerTest -> Tests/Functional/Imagine/Filter/FilterManagerTest

**Comment by makasim** [comment]
yes, you are right! Let's move them

**Comment by wodka** [comment]
@makasim moved the files + squashed the commits :)

**Comment by makasim** [comment]
Thanks for your patience

**Comment by makasim** [comment]
https://github.com/liip/LiipImagineBundle/releases/tag/1.2.2

**Pull request merged**